numba_cuda/numba/cuda/target.py

```diff
@@ -61,6 +62,61 @@ def resolve_value_type(self, val):
 # Implementation
 
 
+class CUDACABIArgPacker(ArgPacker):
```

Comment:
This repeats a significant amount of logic in [`ArgPacker`](https://github.com/numba/numba/blob/c21aa9273ef4298392695b1f4613d29456b53e5c/numba/core/datamodel/packer.py#L60).

Perhaps if we could have the base class encapsulate the calls to `dm.get_arguement_type()`, `dm.as_argument()`, and `dm.from_argument()` in overridable methods, that would eliminate the redundancy, but I'm not sure if that really makes sense or if numba would be receptive to that change.